### PR TITLE
Deprecate 'bases:' field.

### PR DESCRIPTION
**Pull request by monopole**

### CODE REVIEW

File: pkg/commands/kustfile/kustomizationfile_test.go [hunk 1]
```diff
@@ -176,10 +175,11 @@ func TestPreserveCommentsWithAdjust(t *testing.T) {
 
     
 
-# shem qing some comments
+# Some comments
 # This is some comment we should preserve
 # don't delete it
-resources:
+RESOURCES:
```
Comment (Liujingfang1) [inline pkg/commands/kustfile/kustomizationfile_test.go]: Why do you use all capital letters here?

Comment (monopole) [reply pkg/commands/kustfile/kustomizationfile_test.go]: its just a YAML test that was there before.
the conversion lowercases the field names.

**Comment by Liujingfang1** [comment]
Have you thought about allowing `edit fix` to move `bases` to `resources`?

File: pkg/commands/kustfile/kustomizationfile_test.go (hunk 1 above)
Comment (Liujingfang1) [reply pkg/commands/kustfile/kustomizationfile_test.go]: Sounds good.

**Comment by monopole** [comment]
That's implicit in this change:
https://github.com/kubernetes-sigs/kustomize/pull/1189/files#diff-0144cdc603a417dd999ebfba9a9062bcR162-R188

edit fix just writes the file back out after these adjustments.

File: pkg/types/kustomization.go [hunk 2]
```diff
@@ -180,6 +181,10 @@ func (k *Kustomization) FixKustomizationPostUnmarshalling() {
 			k.SecretGenerator[i].EnvSource = ""
 		}
 	}
+	for _, b := range k.Bases {
+		k.Resources = append(k.Resources, b)
+	}
+	k.Bases = nil
```
Comment (monopole) [inline pkg/types/kustomization.go]: This is how `edit fix` moves bases to resources.

**Comment by monopole** [comment]
Any further comments/questions?

**Comment by jbrette** [comment]
Does this mean it will also be necessary to copy paste the vars and configs sections from the base directory to the overlay folders ?

**Comment by monopole** [comment]
@jbrette this only impacts the arrangement of a kustomization file.
Whatever one would have put in a `bases:` field now goes into the `resources:` field.
File layout is not effected.

**Pull request merged**